src/langsmith/deployments-scale.mdx

```diff
@@ -0,0 +1,353 @@
+title: Configure LangSmith Deployment for scale
+sidebarTitle: Deployment
+---
+
+The default configuration for self-hosted LangSmith Deployment can handle substantial read and write load, and you can configure your instance to achieve even higher scale as necessary. This page describes scaling considerations and provides examples to help configure your deployment.
+
+For example configurations, refer to [Example Deployment configurations for scale](#example-deployment-configurations-for-scale).
+
+## Summary
+
+The table below provides an overview comparing different LangSmith Deployment configurations for various load patterns (read requests per second / write requests per second) and standard assistant characteristics (average run execution time of 1 seconds, moderate CPU and memory usage):
+
+|  | **[Low / low](#low-reads-low-writes)** | **[Low / high](#low-reads-high-writes)** | **[High / low](#high-reads-low-writes)** | [Medium / medium](#medium-reads-medium-writes) | [High / high](#high-reads-high-writes) |
+| :--- | :--- | :--- | :--- | :--- | :--- |
+| <Tooltip tip="Number of write requests being processed by the Deployment per second">Write requests per second</Tooltip> | 5 | 5 | 500 | 50 | 500 |
+| <Tooltip tip="Number of read requests being processed by the Deployment per second">Read requests per second</Tooltip> | 5 | 500 | 5 | 50 | 500 |
+| **API replicas**<br />(1 CPU, 2Gi per replica) | 1 (default) | 6 | 10 | 3 | 15 |
+| **Queue replicas**<br />(1 CPU, 2Gi per replica) | 1 (default) | 10 | 1 (default) | 5 | 10 |
+| **N_JOBS_PER_WORKER** | 10 (default) | 50 | 10 | 10 | 50 |
+| **Redis resources** | 2 Gi (default) | 2 Gi (default) | 2 Gi (default) | 2 Gi (default) | 2 Gi (default) |
+| **Postgres resources** | 2 CPU<br />8 Gi (default) | 4 CPU<br />16 Gi memory | 4 CPU<br />16 Gi per replica | 4 CPU<br />16 Gi memory | 8 CPU<br />32 Gi memory |
+
+Below we go into more details about the request handling and scaling patterns as well as provide configuration examples for your LangSmith Deployment.
+
+## Scaling for write load
+
+Write load is primarily driven by the following factors:
+
+- Creation of new runs
+- Creation of new checkpoints during run execution
+- Writing to long term memory
+- Creation of new threads
+- Creation of new assistants
+- Execution of periodic cron jobs
+
+The following systems are primarily responsible for handling write load:
+
+- API server: Handles initial request and persistence of data to the database.
+- Queue: Handles the execution of runs.
+- Redis: Handles the storage of ephemeral data about on-going runs.
+- Postgres: Handles the storage of all data, including run, thread, assistant, cron job, checkpointing and long term memory
+
+### Best practices for scaling the write path
+
+#### Enable the use of queue replicas
+
+By default, the api replica manages the queue and does not use queue replicas. You can enable the use of queue replicas by setting the `queue.enabled` configuration to `true`.
```

Comment:
This depends on deployment scenario - the queue yaml configuration is only if they're self-hosting the helm chart. We should make it obvious who should read (or skip) a given section